akka-persistence/src/main/scala/akka/persistence/journal/WriteJournalBase.scala

```diff
@@ -13,8 +13,8 @@ import akka.persistence.AtomicWrite
 private[akka] trait WriteJournalBase {
   this: Actor ⇒
 
-  lazy val persistence = Persistence(context.system)
-  private def eventAdapters = persistence.adaptersFor(self)
+  val persistence = Persistence(context.system)
+  private val eventAdapters = persistence.adaptersFor(self)
```

Comment:
I think this is Ok. If we'd consider going back to making ad-hoc registrations possible the EventAdapters would need to become mutable I guess?